src/input.rs

```diff
@@ -201,12 +231,22 @@ impl<C: Configuration> IngredientImpl<C> {
         &value.fields
     }
 
-    #[cfg(feature = "salsa_unstable")]
     /// Returns all data corresponding to the input struct.
     pub fn entries<'db>(&'db self, zalsa: &'db Zalsa) -> impl Iterator<Item = &'db Value<C>> {
         zalsa.table().slots_of::<Value<C>>()
     }
 
+    /// Returns the IDs of all input structs of this type.
+    pub fn instances<'db>(
```

Comment:
I guess there's some overlap between `entries` and `instances` and the distinction between the two isn't very clear from their names. `entries` should probably be `values` and `instances` is closer to `keys()`? 

It would be nice if we could avoid having both but obviously, not a big deal